java-sdk/src/main/java/org/sagebionetworks/bridge/sdk/BaseApiCaller.java

```diff
@@ -283,24 +235,11 @@ private void addApplicationHeaders(Request request) {
 
     private <T> T getResponseBodyAsType(HttpResponse response, Class<T> c) {
         String responseBody = getResponseBody(response);
-        try {
-            return mapper.readValue(responseBody, c);
-        } catch (IOException e) {
-            throw new BridgeSDKException("Error message: " + e.getMessage()
-                    + "\nSomething went wrong while converting Response Body JSON into " + c.getSimpleName()
-                    + ": responseBody=" + responseBody, e);
-        }
+        return Utilities.getJsonAsType(responseBody, c);
     }
 
     private <T> T getResponseBodyAsType(HttpResponse response, TypeReference<T> type) {
-        String responseBody = getResponseBody(response);
-        try {
-            return mapper.readValue(responseBody, type);
-        } catch (IOException e) {
-            throw new BridgeSDKException("Error message: " + e.getMessage()
-                    + "\nSomething went wrong while converting Response Body JSON into " + type.getType().getClass().getSimpleName()
-                    + ": responseBody=" + responseBody, e);
-        }
+        return getResponseBodyAsType(response, (Class<T>) type.getClass());
```

Comment:
This is broken. This will deserialize the JSON as a TypeReference, rather than what the type reference is actually referencing. We need a distinct method Class vs TypeReference.